src/hotspot/cpu/aarch64/aarch64_sve_ad.m4

```diff
@@ -853,3 +853,17 @@ BINARY_OP_UNPREDICATED(vsubI, SubVI, S, 4, sve_sub)
 BINARY_OP_UNPREDICATED(vsubL, SubVL, D, 2, sve_sub)
 BINARY_OP_UNPREDICATED(vsubF, SubVF, S, 4, sve_fsub)
 BINARY_OP_UNPREDICATED(vsubD, SubVD, D, 2, sve_fsub)
+
+// vector mask cast
+
+instruct vmaskcast(vReg dst) %{
+  predicate(UseSVE > 0);
+  match(Set dst (VectorMaskCast dst));
```

Comment:
A strict check based on in-type and out-type in predicate could strengthen the pattern.

Reply:
I agree.

Reply:
I'v added the type assertion in the constructor of the `"VectorMaskCastNode"`. See :
```
assert(in_vt->length() == vt->length(), "vector length must match");
assert(type2aelembytes(in_vt->element_basic_type()) == type2aelembytes(vt->element_basic_type()), "element size must match");
```
That's why I didn't add the check in predicate. So do you think it's enough?

Reply:
IMO AD instructions should also be accompanied either by predicates or asserts which stress the conditions for the no-op implementation.

Reply:
OK, thanks! I will add the type restrict in the predicate.